### PR TITLE
UIPFPOL-74 Adjust location filter to support cross-tenant search

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://folio-org.atlassian.net/browse/UIPFPOL-74

## Approach

Since the filter is now a multi-select, it requires passing the current value of the filter (which is an array), not just one element from it. Also make it possible to pass the `crossTenant` flag, which will allow searching for locations in different tenants.

--

Should be merged after: 
- https://github.com/folio-org/stripes-acq-components/pull/775

Should be merged before: 
- https://github.com/folio-org/ui-orders/pull/1602

## Screenshot

https://github.com/folio-org/ui-plugin-find-po-line/assets/88109087/2094db93-7798-4803-91ce-58bc75077075

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by NikitaSedyx — APPROVED** [review]

File: FindPOLine/FindPOLine.js [hunk 1]
```diff
@@ -58,7 +58,12 @@ const resultsFormatter = {
 
 const INIT_PAGINATION = { limit: PLUGIN_RESULT_COUNT_INCREMENT, offset: 0 };
 
-const FindPOLine = ({ addLines, isSingleSelect, ...rest }) => {
+const FindPOLine = ({
+  addLines,
+  crossTenant = false,
```
Comment (alisher-epam) [inline FindPOLine/FindPOLine.js:63]: THe default value can be moved to `FindPOLine.defaultProps`

File: FindPOLine/OrderLinesFilters.js [hunk 2]
```diff
@@ -34,7 +32,14 @@ import { getDateRangeValueAsString } from './utils';
 
 const applyFiltersAdapter = (applyFilters) => ({ name, values }) => applyFilters(name, values);
 
-export function OrderLinesFilters({ activeFilters, applyFilters, disabled, funds = [], materialTypes = [] }) {
+export function OrderLinesFilters({
+  activeFilters,
+  applyFilters,
+  crossTenant = false,
+  disabled = false,
```
Comment (alisher-epam) [inline FindPOLine/OrderLinesFilters.js:39]: move to `OrderLinesFilters.defaultProps`

File: FindPOLine/FindPOLine.js (hunk 1 above)
Comment (usavkov-epam) [reply FindPOLine/FindPOLine.js:63]: `defaultProps` is about to be deprecated in the next major version of the React https://github.com/folio-org/ui-receiving/pull/521

File: FindPOLine/OrderLinesFilters.js (hunk 2 above)
Comment (usavkov-epam) [reply FindPOLine/OrderLinesFilters.js:39]: https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-proptypes-and-defaultprops

**Review by alisher-epam — APPROVED** [review]

**Pull request merged**